src/LegacyRecord.php

```diff
@@ -267,7 +209,7 @@ public function getCollectionName(): string
             throw new NoCollectionNameException();
         }
 
-        return $this->getSchema()->collection;
+        return $this->collection ?: $this->getSchema()->collection;
```

Comment:
```suggestion
        return $this->getSchema()->collection;
```
I think it's better to keep the actual implementation because `getSchema` function can return a schema from fields too.